### PR TITLE
CFamily: improve messages for unsupported options

### LINKED ISSUE #1841: Improve the handling of compiler unsupported options
### Description
Currently, when an unsupported compiler option is provided we see a stacktrace in the output:
image
This is usually interpreted by the users as SonarLint is failing and is not working as expected. While in reality, this is not the case. We intentionally don't support these options (like "/std:c++latest" and "/ZW").

### Proposed solution:
Log the message when an option is not supported without the stacktrace.

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #1841

### CODE REVIEW

File: src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs [hunk 1]
```diff
@@ -128,25 +128,6 @@ public void File_RelativeAndAbsolutePath(string input, string expected)
             req.File.Should().Be(expected);
         }
 
-        [TestMethod]
-        [DataRow("/ZW")]
-        [DataRow("/clr")]
-        public void Should_Skip_Cx_And_Cli(string option)
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs:134]: Merged into a new test below that tests all of the unsupported options.

File: src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs [hunk 2]
```diff
@@ -646,9 +627,32 @@ public void Microsoft_Extensions()
...
+                    Env = new List<string>(),
+                    Cmd = new List<string>{ "cl.exe", option, "anyfile.txt" }
+                }
+            });
+
+            act.Should().ThrowExactly<NotSupportedException>().And.Message.Should().Be(expectedMessage);
+        }
+
+        [TestMethod]
+        [DataRow("@foo", "unexpanded response file")]
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs:652]: WIP: checking with Abbas whether this should be "invalid" or "unsupported"

**Comment by duncanp-sonar** [comment]
Examples of new output:
image

File: src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs (hunk 2 above)
Comment (duncanp-sonar) [reply src/Integration.Vsix.UnitTests/CFamily/MsvcDriverTest.cs:652]: Done: Abbas suggests keeping this as invalid for now.

File: src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs [hunk 3]
```diff
@@ -196,6 +196,12 @@ public void RequestAnalysis(string path, string charset, IEnumerable<AnalysisLan
                             analyzerOptions, cancellationToken),
                     analysisTimeout);
             }
+            catch (NotSupportedException ex)
+            {
+                // Display a simple user-friendly message for options we know are not supported.
+                // See https://github.com/SonarSource/sonarlint-visualstudio/pull/2212
+                logger.WriteLine($"Analysis error: {ex.Message}");
```
Comment (rita-g-sonarsource) [inline src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs:203]: Perhaps we should rephrase it not as an error, e.g. "Unsupported analysis option: ..."
WDYT?

**Review by rita-g-sonarsource — APPROVED** [review]

File: src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs (hunk 3 above)
Comment (duncanp-sonar) [reply src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs:203]: That doesn't make it clear that the analysis hasn't taken place (could be interpreted as meaning the setting is not taken into account).

I'll change it to "Unable to analyze: ..."

**Pull request merged**